libraries/monitor/src/logger.js

```diff
@@ -0,0 +1,144 @@
+const os = require('os');
+const assert = require('assert');
+const chalk = require('chalk');
+const stringify = require('fast-json-stable-stringify');
+
+const LEVELS = {
+  emerg: 0,
+  alert: 1,
+  crit: 2,
+  err: 3,
+  warning: 4,
+  notice: 5,
+  info: 6,
+  debug: 7,
+};
+
+const LEVELS_REVERSE = [
+  chalk.red.bold('EMERGENCY'),
+  chalk.red.bold('ALERT'),
+  chalk.red.bold('CRITICAL'),
+  chalk.red('ERROR'),
+  chalk.yellow('WARNING'),
+  chalk.blue('NOTICE'),
+  chalk.green('INFO'),
+  chalk.magenta('DEBUG'),
+];
+
+/*
+ * Implements the mozlog standard as defined in
+ * https://wiki.mozilla.org/Firefox/Services/Logging
+ *
+ * We can consider supporting extra logging standards
+ * later if we want.
+ */
+class Logger {
+  constructor({name, level, pretty=false, enable=true, destination=process.stdout, metadata=null}) {
+    assert(name, 'Must specify Logger name.');
+
+    this.name = name;
+    this.destination = destination;
+    this.pretty = pretty;
+    this.enable = enable;
+    this.metadata = Object.keys(metadata).length > 0 ? metadata : null;
+    this.pid = process.pid;
+    this.hostname = os.hostname();
+
+    level = level.trim();
+    assert(LEVELS[level] !== undefined, `Error levels must correspond to syslog severity levels. ${level} is invalid.`);
+    this.level = LEVELS[level];
+  }
+
+  _log(level, type, fields) {
+    if (!this.enable || level > this.level) {
+      return;
+    }
+
+    if (fields === undefined) {
+      fields = type;
+      type = 'generic';
+    }
+    if (typeof fields === 'number') {
+      fields = {val: fields};
+    }
+    if (typeof fields === 'string') {
+      fields = {msg: fields};
+    }
```

Comment:
My plan is to have it be used for replacing `debug` easily and the like. I will document it.

Also just wanted there to be very few cases in which this would crash.